sway-core/src/semantic_analysis/ast_node/declaration/declaration.rs

```diff
@@ -70,7 +70,12 @@ impl ty::TyDeclaration {
                         type_ascription,
                         type_ascription_span,
                     }));
-                ctx.namespace.insert_symbol(name, typed_var_decl.clone());
+                check!(
+                    ctx.namespace.insert_symbol(name, typed_var_decl.clone()),
+                    return err(warnings, errors),
+                    warnings,
+                    errors
+                );
```

Comment:
Ahh I wonder how we missed these in the past :thinking: Perhaps we should decorate `CompileResult` with `#[must_use]` so that we are forced to handle results like this, or explicitly ignore them (like `std::result::Result`)? Not the responsibility of this PR of course, just a thought!

Reply:
Hmm that's very interesting. I'll create an issue. That being said, if we end up implementing https://github.com/FuelLabs/sway/issues/2734, then we won't have that particular issue I think.